### PR TITLE
FLUID-5903 Working implementation of "grade budding support" - docs pull required

### PR DESCRIPTION
**Pull request by amb26**
This pull request depends on #754

### CODE REVIEW

**Comment by jobara** [comment]
@amb26 is the docs PR forthcoming?

**Comment by jobara** [comment]
@amb26 DataBinding test 43 "FLUID-5659 repeated relay test: relay test" fails, and also halts the all-tests run. 

Error:
Assertion failure (see console.log for expanded message): Could not decode fixture record ,e, as a valid fixture type - must contain key chosen from func, funcName, event, changeEvent, task, jQueryTrigger, jQueryBind
Source: 	
    at jqUnit.failureHandler (http://localhost:8888/infusion/tests/test-core/jqUnit/js/jqUnit.js:33:19)
    at Object.fire (http://localhost:8888/infusion/src/framework/core/js/Fluid.js:1493:40)
    at Object.fluid.fail (http://localhost:8888/infusion/src/framework/core/js/Fluid.js:212:32)
    at Object.fluid.test.decodeFixture (http://localhost:8888/infusion/tests/test-core/utils/js/IoCTestUtils.js:420:19)
    at Object.that.decode (http://localhost:8888/infusion/tests/test-core/utils/js/IoCTestUtils.js:525:46)
    at Object.fluid.test.sequenceExecutor (http://localhost:8888/infusion/tests/test-core/utils/js/IoCTestUtils.js:538:14)
    at Object.testFunc (http://localhost:8888/infusion/tests/test-core/utils/js/IoCTestUtils.js:630:32)
    at Test.run (http://localhost:8888/infusion/tests/lib/qunit/js/qunit.js:207:18)
Died on test #2     at Object.asyncTest (http://localhost:8888/infusion/tests/lib/qunit/js/qunit.js:405:9)
    at http://localhost:8888/infusion/tests/test-core/utils/js/IoCTestUtils.js:648:29
    at Object.fluid.each (http://localhost:8888/infusion/src/framework/core/js/Fluid.js:512:17)
    at Object.fluid.test.processTestCase (http://localhost:8888/infusion/tests/test-core/utils/js/IoCTestUtils.js:622:15)
    at http://localhost:8888/infusion/tests/test-core/utils/js/IoCTestUtils.js:671:24
    at Object.fluid.each (http://localhost:8888/infusion/src/framework/core/js/Fluid.js:512:17)
    at Object.fluid.test.processTestCaseHolder (http://localhost:8888/infusion/tests/test-core/utils/js/IoCTestUtils.js:664:15)
    at http://localhost:8888/infusion/tests/test-core/utils/js/IoCTestUtils.js:720:24: Assertion failure - check console for more details: Could not decode fixture record e as a valid fixture type - must contain key chosen from func, funcName, event, changeEvent, task, jQueryTrigger, jQueryBind
Source: 	
Error: Assertion failure - check console for more details: Could not decode fixture record e as a valid fixture type - must contain key chosen from func, funcName, event, changeEvent, task, jQueryTrigger, jQueryBind
    at fluid.FluidError (http://localhost:8888/infusion/src/framework/core/js/Fluid.js:173:26)
    at fluid.builtinFail (http://localhost:8888/infusion/src/framework/core/js/Fluid.js:197:15)
    at Object.fire (http://localhost:8888/infusion/src/framework/core/js/Fluid.js:1493:40)
    at Object.fluid.fail (http://localhost:8888/infusion/src/framework/core/js/Fluid.js:212:32)
    at Object.fluid.test.decodeFixture (http://localhost:8888/infusion/tests/test-core/utils/js/IoCTestUtils.js:420:19)
    at Object.that.decode (http://localhost:8888/infusion/tests/test-core/utils/js/IoCTestUtils.js:525:46)
    at Object.fluid.test.sequenceExecutor (http://localhost:8888/infusion/tests/test-core/utils/js/IoCTestUtils.js:538:14)
    at Object.testFunc (http://localhost:8888/infusion/tests/test-core/utils/js/IoCTestUtils.js:630:32)
    at Test.run (http://localhost:8888/infusion/tests/lib/qunit/js/qunit.js:207:18)
    at http://localhost:8888/infusion/tests/lib/qunit/js/qunit.js:365:10

**Comment by amb26** [comment]
Thanks for this catch, @jobara - tests now fixed up

**Comment by jobara** [comment]
@amb26 when do you expect the docs pr to be ready?

File: tests/framework-tests/core/js/DataBindingTests.js [hunk 1]
```diff
@@ -2242,7 +2242,7 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
                 options: {
                     moduleSource: {
                         funcName: "fluid.tests.fluid5659source",
-                        args: "{fluid5659relay}.options.lang"
+                        args: ["{fluid5659relay}.options.lang"]
```
Comment (jobara) [inline tests/framework-tests/core/js/DataBindingTests.js]: @amb26 is there a new requirement now that args must be in an array?

Comment (amb26) [inline tests/framework-tests/core/js/DataBindingTests.js]: It is actually the reverse - it used to be that args being an array was the only possibility - and hence binding it to an array value would represent the whole of args. This was really a dispatching bug. Now it is possible that args is a single argument, and hence to remove the ambiguity we need to explicitly represent the whole argument array - that is, "lang" is an array, and the function fluid5659source expects a single argument which is an array.

**Comment by cindyli** [comment]
Merged at ff72b53d80c2c11c360e57e8f62e50dbca337875

**Comment by cindyli** [comment]
@amb26, thanks for the implementation. Don't forget to send a pull request for updating the documentation when you have a chance.

**Comment by amb26** [comment]
@jobara - Here is the matching docs pull: https://github.com/fluid-project/infusion-docs/pull/103

**Pull request merged**